src/wallet/test/walletdb_tests.cpp

```diff
@@ -29,5 +29,104 @@ BOOST_AUTO_TEST_CASE(walletdb_readkeyvalue)
+    std::shared_ptr<CWallet> wallet = std::make_shared<CWallet>(nullptr, "", std::move(database));
+    wallet->SetWalletFlag(WALLET_FLAG_DESCRIPTORS);
+    {
+        LOCK(wallet->cs_wallet);
+        WalletBatch batch(wallet->GetDatabase());
+
+        // Write some basic wallet data
+        batch.WriteWalletFlags(WALLET_FLAG_DESCRIPTORS);
+        BOOST_CHECK(batch.TxnBegin());
+        BOOST_CHECK(batch.TxnCommit());
+
+        // Load the wallet to establish baseline
+        DBErrors load_result = batch.LoadWallet(wallet.get());
+        BOOST_CHECK(load_result == DBErrors::LOAD_OK);
+    }
+
+    // Now test with read-only database
+    MockableData wallet_data = dynamic_cast<MockableDatabase&>(wallet->GetDatabase()).m_records;
+    auto readonly_database = CreateMockableWalletDatabase(wallet_data, true);
+    BOOST_CHECK(readonly_database->IsReadOnly());
+
+    std::shared_ptr<CWallet> readonly_wallet = std::make_shared<CWallet>(nullptr, "", std::move(readonly_database));
+    readonly_wallet->SetWalletFlag(WALLET_FLAG_DESCRIPTORS);
+
+    {
+        LOCK(readonly_wallet->cs_wallet);
+        WalletBatch readonly_batch(readonly_wallet->GetDatabase());
+
+        // LoadWallet should succeed on read-only database
+        DBErrors load_result = readonly_batch.LoadWallet(readonly_wallet.get());
+        BOOST_CHECK(load_result == DBErrors::LOAD_OK);
+
+        // The readonly wallet should have the same flags as the original
+        BOOST_CHECK(readonly_wallet->IsWalletFlagSet(WALLET_FLAG_DESCRIPTORS));
+    }
+}
+
+BOOST_AUTO_TEST_CASE(walletdb_readonly_batch_operations)
+{
+    // Test WalletBatch operations with read-only database
+    MockableData initial_data;
+
+    // Pre-populate with some wallet-like data
+    DataStream key_stream, value_stream;
+    key_stream << std::make_pair(DBKeys::VERSION, std::string{});
+    value_stream << CLIENT_VERSION;
+    initial_data[SerializeData(key_stream.begin(), key_stream.end())] = SerializeData(value_stream.begin(), value_stream.end());
+
+    key_stream.clear();
+    value_stream.clear();
+    key_stream << std::make_pair(DBKeys::FLAGS, std::string{});
+    value_stream << uint64_t{WALLET_FLAG_DESCRIPTORS};
+    initial_data[SerializeData(key_stream.begin(), key_stream.end())] = SerializeData(value_stream.begin(), value_stream.end());
+
+    auto readonly_database = CreateMockableWalletDatabase(initial_data, true);
+
+    // Test that we can create a batch and perform read operations
+    std::unique_ptr<DatabaseBatch> batch = readonly_database->MakeBatch();
+
+    // Reading should work
+    int version;
+    BOOST_CHECK(batch->Read(std::make_pair(DBKeys::VERSION, std::string{}), version));
+    BOOST_CHECK_EQUAL(version, CLIENT_VERSION);
+
+    uint64_t flags;
+    BOOST_CHECK(batch->Read(std::make_pair(DBKeys::FLAGS, std::string{}), flags));
+    BOOST_CHECK_EQUAL(flags, WALLET_FLAG_DESCRIPTORS);
+}
+
+BOOST_AUTO_TEST_CASE(walletdb_readonly_database_property)
```

Comment:
In 6189a20491f71e3a6f499d2a510b38bb914e00d9 "test: Add tests for readonly database access"

I don't think that this test needs to be it's own test case. I think all of the tests added in this file can be combined into a single `walletdb_readonly_database` test